packages/expo-notifications/README.md

```diff
@@ -1 +1,367 @@
+- **development (_boolean_)** -- Makes sense only on iOS, where there are two push notification services: sandbox and production. This defines whether the push token is supposed to be used with the sandbox platform notification service. Defaults to [`Application.getIosPushNotificationServiceEnvironmentAsync()`](https://docs.expo.io/versions/latest/sdk/application/#applicationgetiospushnotificationserviceenvironmentasync) exposed by `expo-application` or `false`. Most probably you won't need to customize that.
+- **baseUrl (_string_)** -- Base URL upon which the function will build a URL to which it will send the device push token. If `url` is defined, this option won't have any effect. Defaults to a production Expo backend URL. Most probably you won't need to change that.
+- **url (_string_)** -- Endpoint URL to which the function will send the device push token. Most probably you won't need to customize that. Setting `url` overrides `baseUrl`.
+
+#### Returns
+
+Returns a `Promise` that resolves to an object with the following fields:
+
+- **type (_string_)** -- Always `expo`.
+- **data (_string_)** -- The push token as a string.
+
+#### Examples
+
+##### Fetching the Expo push token and uploading it to a server
+
+```ts
+import Constants from 'expo-constants';
+import * as Notifications from 'expo-notifications';
+
+export async function registerForPushNotificationsAsync(userId: string) {
+  let experienceId = undefined;
+  if (!Constants.manifest) {
+    // Absence of the manifest means we're in bare workflow
+    experienceId = '@username/example';
+  }
+  const expoPushToken = await Notifications.getExpoPushTokenAsync({
+    experienceId,
+  });
+  await fetch('https://example.com/', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      userId,
+      expoPushToken,
+    }),
+  });
+}
+```
+
+### `getDevicePushTokenAsync(): DevicePushToken`
+
+Returns a native APNS, FCM token or a [`PushSubscription` data](https://developer.mozilla.org/en-US/docs/Web/API/PushSubscription) that can be used with another push notification service.
+
+#### Returns
+
+A `Promise` that resolves to an object with the following fields:
+
+- **type (_string_)** -- Either `ios`, `android` or `web`.
+- **data (_string_ or _object_)** -- Either the push token as a string (for `type == "ios" | "android"`) or an object conforming to the type below (for `type == "web"`):
+  ```ts
+  {
+    endpoint: string;
+    keys: {
+      p256dh: string;
+      auth: string;
+    }
+  }
+  ```
+
+### `addPushTokenListener(listener: PushTokenListener): Subscription`
+
+In rare situations a push token may be changed while the app is running. When a token is rolled, the old one becomes invalid and sending notifications to it will fail. A push token listener will let you handle this situation gracefully.
```

Comment:
```suggestion
In rare situations a push token may be changed by the push notification service while the app is running. When a token is rolled, the old one becomes invalid and sending notifications to it will fail. A push token listener will let you handle this situation gracefully by registering the new token with your backend right away.
```